packages/web-console/src/scenes/Import/ImportCSVFiles/utils.ts

```diff
@@ -0,0 +1,49 @@
+import { SchemaColumn } from "components/TableSchemaDialog/types"
+
+export const isGeoHash = (type: string) => type.startsWith("GEOHASH")
+
+export const extractPrecionFromGeohash = (geohash: string) => {
+  const regex = /\(([^)]+)\)/g
+  const matches = regex.exec(geohash)
+  if (matches && matches.length > 1) {
+    return matches[1]
+  }
+  return ""
+}
+
+export const mapColumnTypeToUI = (type: string) => {
+  if (isGeoHash(type)) {
+    return "GEOHASH"
+  } else return type.toUpperCase()
+}
+
+export const mapColumnTypeToQuestDB = (column: SchemaColumn) => {
+  if (column.type === "GEOHASH") {
+    return {
+      ...column,
+      type: `GEOHASH(${column.precision})`,
+    }
+  }
+  return column
+}
+
+const str = () =>
+  (
+    "00000000000000000" + (Math.random() * 0xffffffffffffffff).toString(16)
+  ).slice(-16)
+
+export const uuid = () => {
```

Comment:
is this how you generate a random uuid in JS? 😱

Reply:
@jerrinot Nope, there are libraries to do so, but they all require `crypto` to be available in the system - so either HTTPS or localhost. We had users that use neither (just plain `http://`) and the system failed for them. For that reason I have just used a simple good-enough method to generate a random identifier (it's not persisted anywhere). 
I could use `new Date().getTime()` as an alternative, but you can drop multiple files to be uploaded at the same time, so there is a possibility for a trouble, I think 🙂

Reply:
thanks! 
I agree it's unimportant here, I was just curious.